stdlib/asm/math/poly512.masm

```diff
@@ -0,0 +1,6682 @@
+use.std::math::ntt512
+use.std::math::u64
+
+# Given two consecutive words on stack, this routine performs 
+# element wise multiplication, while keeping resulting single
+# word on stack.
+#
+# Expected stack state looks like
+#
+# [a0, a1, a2, a3, b0, b1, b2, b3]
+#
+# What this routine does is
+#
+# c`i` = a`i` * b`i` mod P | i ∈ [0, 4), P = 2 ^ 64 - 2 ^ 32 + 1
+#
+# Output stack state looks like
+#
+# [c0, c1, c2, c3]
+proc.mul_word
+    movup.4
+    mul
+
+    swap
+    movup.4
+    mul
+    swap
+
+    movup.2
+    movup.4
+    mul
+    movdn.2
+
+    movup.3
+    movup.4
+    mul
+    movdn.3
+end
```

Comment:
I think you could do something like this here and for addition:
```
movup.4
mul
movdn.6
movup.3
mul
movdn.5
movup.2
mul
movdn.4
mul
movdn.3
```
This should save a few cycles.

But also, I wonder if we should just create assembly instructions: `addw`, `mulw` so that people don't have to re-implement these.

Reply:
> I think you could do something like this here and for addition:

Yes, I've replaced both of them with suggested one in commit 322a31e

Reply:
> But also, I wonder if we should just create assembly instructions: `addw`, `mulw` so that people don't have to re-implement these.

I think we can do so, but we can probably defer it to sometime later. Because this is the very first time I've to use this kind of routine, while it looks like this kind of routine will be pretty useful, in other scenarios.